dvc/repo/fetch.py

```diff
@@ -107,3 +119,25 @@ def _fetch(repo, obj_ids, **kwargs):
     except FileTransferError as exc:
         failed += exc.amount
     return downloaded, failed
+
+
+def _fetch_partial_imports(repo, targets, **kwargs):
+    from dvc.stage.exceptions import DataSourceChanged
+
+    downloaded = 0
+    failed = 0
+    for stage in repo.partial_imports(targets, **kwargs):
+        try:
+            stage.run()
```

Comment:
Using `run()`/`sync_import()` here means that this is always a `pull` and not really a `fetch`. The import will be downloaded into the workspace and then saved/committed to cache, but the output will remain "checked out" in the workspace, even if the user ran fetch. 

in this workflow I would not expect `file` to be present in my workspace (it should only be in `.dvc/cache` to be checked out later):
```
$ dvc import-url "azure://test-container/dir/file" --no-download
Importing 'azure://test-container/dir/file' -> 'file'
$ ls
file.dvc  tags
$ dvc fetch
Importing 'azure://test-container/dir/file' -> 'file'
1 file fetched
$ ls
file  file.dvc  tags
```